### DISCUSSION TITLE
Expose state via slots

### DISCUSSION THREAD

**Discussion by BayBreezy** [unovue/reka-ui#439] [Ideas]
Is it possible to expose the open state of some components via slot?

Like they have in headlessui/vue. An example would be for the AccordionTrigger component. in addition to the open/closed state being updated via a data attribute, I think it would be nice if we were able to do something like this

```vue
  <AccordionTrigger
    v-bind="forwarded"
    :class="accordionTriggerStyles({ hover, class: props.class })"
    v-slot="{ open }"
  >
    <slot>{{ title }}</slot>
    <slot name="icon">
      <Icon :name="icon" class="h-4 w-4 text-muted-text transition-transform" />
    </slot>
  </AccordionTrigger>
```

This way we can take the open value and do stuff with it like hide an element or rotate the icon

**Comment by nhedger** [comment]
I second this.

Would be really helpful, especially when dynamically creating components where tracking their state could be a bit harder.

Assuming the maintainers are OK, I'd be able to contribute this feature to some of the components.

**Reply by zernonia** [reply]
Hey @nhedger ! We are already exposing some slot props.. but we might missed out some.. Totally welcome PR for this! 😁

**Reply by nhedger** [reply]
Sounds good, I'll PR something shortly!

**Comment by MangMax** [comment]
I'm also experiencing this issue when I use dropdown menu, I don't know how to get the state, however I see that Accordion does expose the state through the slot.

**Comment by ragulka** [comment]
This would be really beneficial when working with Combobox as well. Right now, there's no way to know if there are any results matching the query, other than having to provide the filtered options myself. 

I see that the `ComoboxEmpty` component [uses the root context to determine if it's empty](https://github.com/radix-vue/radix-vue/blob/6dbf5805da20f585536fc8534a06a25956002dfa/packages/radix-vue/src/Combobox/ComboboxEmpty.vue#L17-L18), but this state isn't exposed, nor can we access the root context ourself.

Ideally, this state would be exposed as slot props in various components like `ComoboxContent`, etc.
